src/app/infrastructure/RequestAction.ts

```diff
@@ -0,0 +1,84 @@
+import { ThunkAction, RootState, ThunkExtra } from "../store";
+import { AnyAction } from "redux";
+
+export interface RequestActionState<T> {
+    isBusy: boolean;
+    error?: string;
+    data: T;
```

Comment:
Do we need a success attribute here? or is that indicated by the absence of an error?

Reply:
good question.  I gave this some serious thought and wondered what we'd need a success flag / attribute for.  I figured we'd only use it for success toasts or messages that pop up after you do something.  So instead of having a flag I put in hooks to override the default dispatching for success, fail and begin:

https://github.com/bcgov/jag-shuber-frontend/pull/28/files#diff-836d9d2a6cd53ce38ad1662522968742R55

Reply:
Awesome - I figured you had thought about it :+1: